### DISCUSSION TITLE
Help with Custom Endpoints

### DISCUSSION THREAD

**Discussion by laurensiusadi** [directus/directus#6423] [Archive]
So I want to make a custom endpoint, like a getter of some sort. It does some calculations with conditionals. So I want to do a get (findOne by id) from 2 different collections one called 'products' the other called 'brands'.

The [example on docs](https://docs.directus.io/guides/api-endpoints/#full-example) isn't helpful at all. I don't understand what `service`, `schema` and `accountability` is. And how do I actually use the endpoint with REST (Postman)?

**Comment by laurensiusadi** [comment]
Apparently I only need to change the `readByQuery` to `readOne` and put the id there.
But now I'm getting errors `"You don't have permission to access this."`

Permission to read on both collections is enabled for public.
What am I missing?

**Reply by laurensiusadi** [reply]
Turns out I'm accessing the wrong product id, which not exist in DB. But it says no permission instead of not exist.

**Reply by rijkvanzanten** [reply]
> But it says no permission instead of not exist.

This is for security reasons, we don't want people to be able to scrape the available records in the project by brute forcing 404s in the endpoints 👍🏻
